src/p2p/mod.rs

```diff
@@ -52,7 +52,7 @@ pub async fn create_swarm<TIpfsTypes: IpfsTypes>(
     // Set up an encrypted TCP transport over the Mplex protocol.
     let transport = transport::build_transport(options.keypair.clone());
 
-    let swarm_span = ipfs.0.span.clone();
+    let swarm_span = ipfs.span.clone();
```

Comment:
I was thinking we maybe should do a `tracing::trace_span!(parent: &ipfs.span, "swarm")` "pass" over the current tracing usage as I am not sure what does this Clone exactly do, but at least it's no longer public member so I am fine with this.